listener/listener.go

```diff
@@ -481,3 +538,28 @@ func (l *Listener) setLSN(lsn uint64) {
 
 	l.lsn = lsn
 }
+
+func (l *Listener) processPipelineResult(raw interface{}) error {
+	if raw == nil {
+		return nil
+	}
+
+	switch v := raw.(type) {
+	case error:
+		l.log.Error("pipeline outcome error", slog.Any("err", v))
+		return v
+	case uint64:
+		if v > l.readLSN() {
+			if err := l.AckWalMessage(v); err != nil {
+				l.monitor.IncProblematicEvents(problemKindAck)
+				return fmt.Errorf("ack: %w", err)
+			}
+
+			l.log.Debug("ack WAL message", slog.Uint64("lsn", l.readLSN()))
+		}
+	default:
+		panic(fmt.Sprintf("unexpected type in pipeline: %T", v))
```

Comment:
This feels a bit smelly, if we were using typed channels stuff like this wouldn't be necessary.